modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/notification/term/evaluator/CommerceOrderAddressNotificationTermEvaluator.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.commerce.internal.notification.term.evaluator;
+
+import com.liferay.account.model.AccountEntry;
+import com.liferay.commerce.model.CommerceAddress;
+import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.service.CommerceOrderLocalService;
+import com.liferay.notification.term.evaluator.NotificationTermEvaluator;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Country;
+import com.liferay.portal.kernel.model.Region;
+import com.liferay.portal.kernel.security.permission.PermissionCheckerFactory;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.service.RoleLocalService;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Map;
+
+/**
+ * @author Balázs Breier
+ */
+public class CommerceOrderAddressNotificationTermEvaluator
+	implements NotificationTermEvaluator {
+
+	public CommerceOrderAddressNotificationTermEvaluator(
+		ModelResourcePermission<AccountEntry>
+			accountEntryModelResourcePermission,
+		CommerceOrderLocalService commerceOrderLocalService,
+		ObjectDefinition objectDefinition,
+		PermissionCheckerFactory permissionCheckerFactory,
+		RoleLocalService roleLocalService, UserLocalService userLocalService) {
+
+		_accountEntryModelResourcePermission =
+			accountEntryModelResourcePermission;
+		_commerceOrderLocalService = commerceOrderLocalService;
+		_objectDefinition = objectDefinition;
+		_permissionCheckerFactory = permissionCheckerFactory;
+		_roleLocalService = roleLocalService;
+		_userLocalService = userLocalService;
+	}
+
+	@Override
+	public String evaluate(Context context, Object object, String termName)
+		throws PortalException {
+
+		if (!(object instanceof Map) ||
+			!"CommerceOrder".equalsIgnoreCase(
+				_objectDefinition.getShortName())) {
+
+			return termName;
+		}
+
+		Map<String, Object> termValues = (Map<String, Object>)object;
+
+		return _getAddress(termName, termValues);
+	}
+
+	private String _getAddress(String termName, Map<String, Object> termValues)
+		throws PortalException {
+
+		CommerceOrder commerceOrder =
+			_commerceOrderLocalService.getCommerceOrder(
+				GetterUtil.getLong(termValues.get("id")));
+
+		String billingAddressPrefix = "[%COMMERCEORDER_BILLING_ADDRESS_";
+		String shippingAddressPrefix = "[%COMMERCEORDER_SHIPPING_ADDRESS_";
```

Comment:
Hey @brbalazs can we move these two at the beginning of the method and do a check? If the termName is not valid (doesn't start with what we need) returns immediately the termName. In this way we avoid to fetch the order from the database if we don't need it

Reply:
sorry to didn't catch this before